tests/phpunit/integration/Core/Admin/DashboardTest.php

```diff
@@ -21,15 +21,210 @@
 class DashboardTest extends TestCase {
 
 	public function test_register() {
+		global $wp_meta_boxes;
+		$this->get_widget_markup();
+
+		// Check that the dashboard widget was registered
+		$this->assertArrayHasKey( 'google_dashboard_widget', $wp_meta_boxes['dashboard']['normal']['core'] );
+
+		// Check that expected assets are enqueued
+		$this->assertTrue( wp_script_is( 'googlesitekit-wp-dashboard', 'enqueued' ) );
+		$this->assertTrue( wp_style_is( 'googlesitekit-wp-dashboard-css', 'enqueued' ) );
+	}
+
+	/**
+	 * @group dashboard_widget
+	 */
+	public function test_render_googlesitekit_wp_dashboard_loading_container() {
+		$output = $this->get_widget_markup();
+
+		$dom = new \DOMDocument();
+		$dom->loadHTML( $output );
+
+		$this->assertNotEmpty( $dom->getElementById( 'js-googlesitekit-wp-dashboard' ) );
+
+		$xpath = new \DOMXPath( $dom );
+		$query = '//div[contains(concat(" ", normalize-space(@class), " "), " googlesitekit-wp-dashboard ") and contains(concat(" ", normalize-space(@class), " "), " googlesitekit-wp-dashboard-loading ")]';
+
+		$elements = $xpath->query( $query );
+
+		$this->assertEquals( 1, $elements->length );
+	}
+
+	/**
+	 * @group dashboard_widget
+	 */
+	public function test_render_googlesitekit_wp_dashboard_only_search_console_connected() {
+		// Only set the search console module as connected.
+		$active_modules = array(
+			'search-console' => $this->getMockBuilder( 'Google\Site_Kit\Core\Modules' )
+				->disableOriginalConstructor()
+				->setMethods( array( 'get_active_modules' ) )
+				->getMock(),
+		);
+
+		$active_modules['search-console']
+			->method( 'get_active_modules' )
+			->willReturn(
+				array(
+					'search-console' => (object) array(
+						'is_connected' => function () {
+							return true;
+						},
+					),
+				)
+			);
```

Comment:
Please update this test to avoid mocking these methods, and take a more integrated approach instead.

Reply:
Removed this mocking. Instead now passing the module instance to `Dashboard`